archive/c/c-sharp/roman-numeral.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+
+namespace SamplePrograms
+{
+    public class RomanNumeral
+    {
+        private static readonly Dictionary<char, int> RomanDecMapping = new Dictionary<char, int>()
+        {
+            ['M'] = 1000,
+            ['D'] = 500,
+            ['C'] = 100,
+            ['L'] = 50,
+            ['X'] = 10,
+            ['V'] = 5,
+            ['I'] = 1,
+        };
+
+        private static int RomanToDecimal(string roman, int total=0)
+        {
+            var romanArray = roman.ToCharArray();
+            if (romanArray.Length == 1)
+                return total + RomanDecMapping[romanArray[0]];
+
+            var romanVal = RomanDecMapping[romanArray[0]];
+            var nextRomanVal = RomanDecMapping[romanArray[1]];
```

Comment:
I'll need some opinions from @TheRenegadeCoder/core on this, but I'm pretty sure this causes an exception if the character is not a roman digit. Do we want to introduce a sanity check here, albeit at a expense of brevity?

Reply:
I'm glad you pointed this out. I had to revisit the [original solution which was in Python](https://github.com/TheRenegadeCoder/sample-programs/blob/master/archive/p/python/roman-numeral.py). After digging into it a bit, I noticed that for invalid symbols, we return -1. However, we never actually verify that the input is valid.

As a result, input like `XT` will result in -9 which isn't true. In fact, this is an error, and I think it should be addressed in both solutions as well as updated in the rules.

How we address it though may prove to be too complicated though, so let's see what we can come up with.

Reply:
Hey, I just updated the roman numeral docs. Let me know how you like them for addressing this issue. #595